_posts/2016-01-30-iOS-Junior-Interviews.markdown

```diff
@@ -0,0 +1,300 @@
+
+There is no catch-all solid advice for interviews, but I can give you some of the things we talk about in Artsy. 
+
+We look for [T shaped](https://en.wikipedia.org/wiki/T-shaped_skills) engineers, even in juniors. This means someone who has experience in a pretty wide net of things, but that they have a solid focus. This could be building their APIs for apps, writing their own blog, designing their website, automating some regular tasks or using technology for art.
+
+As a junior, the employer is looking for growth potential. Looking for people who, in the right environment could really thrive. However it's important to note, an interview only lets you know how well someone interviews. The greatest programmers can choke on interviews.
+
+##### How I interview Juniors
+
+One thing I can tell you concretely though, is what I do. My interviews with juniors come in three parts:
+
+- A Chat
+
+  I want to get someone comfortable with the interview, I'll have taken points from looking over the portfolio to explore. It will be a pretty one-sided conversation with me only trying to provide points to spring off. "So tell me about when you did x". 
+
+- Pair Programming
+
+  I want to get a sense of how you use a computer and act under mentorship. So I do one of two things, depending on how work has been the last week. If there's been a pull request which seem small and contained enough from our team on one of our apps. I will pair on re-creating the pull request from scratch, without letting it be known that the PR had already been built and accepted. If there isn't, we'll take a pre-built broken app and fix it.
+
+  I want to see things like; do you use Xcode shortcuts? Can you explain the code you're looking at? If I offer some advice, do you use it? Can you present ideas when we're figuring out an abstraction? Can you identify where problematic behaviour lies?
+
+- Learn you a thing
+
+  Alright, so I've been building up the applicant's confidence and now it's time to bring that all down. Sorry. I have been paying attention to finding out what the applicant doesn't know. Then I start asking questions about this. It doesn't really matter what the topic is, it could be threading, view controller lifecycle, code coupling, dynamic vs static dispatch, whatever. The point is that I want to understand how someone learns during a discussion. So the conversation tends to switch around, where-in I lead a conversation on the topic - try to lay foundations then experiment with questions that should require an understanding of the topic.
+
+Here's [a long writeup](https://medium.com/@jon.lazar/my-approach-to-learning-ios-fccf943aead#.ei4yvc38k), with a lot of depth (and places to jump off from) from someone who has just come out of a mock interview with me. Thanks Jon.
+
+None of this process aims to be adversarial, if someone has got this far, I really want to have a sense of how much time and attention will be required to give them some independence.
```

Comment:
Super interesting approach. I actually hope this part is read by people looking to hire juniors moreso than juniors, since it's a great way to evaluate and I don't know if it's widespread.